File: Python/1_twoSum.py
'''
Given an array of integers nums and an integer target, return indices of the two numbers such that they add up to target.

You may assume that each input would have exactly one solution, and you may not use the same element twice.

You can return the answer in any order.
'''


class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        second_num = dict()
        for i, each in enumerate(nums):
            if each in second_num:
                return [second_num[each], i]
            else:
                second_num[target - each] = i

File: Golang/1_twoSum.go
/*
    Given an array of integers nums and an integer target, return indices of the two numbers such that they add up to target.
    You may assume that each input would have exactly one solution, and you may not use the same element twice.
    You can return the answer in any order.
*/

func twoSum(nums []int, target int) []int {
    secondNum := make(map[int]int)
    for index, num := range nums {
        _, ok := secondNum[num]
        if ok {
            return []int{secondNum[num], index}
        } else {
            secondNum[target - num] = index
        }
    }
    return []int{}
}